### DISCUSSION TITLE
Expo Web throwing error "ReferenceError: window is not defined"

### DISCUSSION THREAD

**Discussion by John7bigo** [supabase/supabase#25909] [Questions]
I've been ecountering this error "ReferenceError: window is not defined" when using the "import AsyncStorage from '@react-native-async-storage/async-storage';" on my project and running the Expo Web version. I do not have the issue with when running the IOS or Android with the expo app. If I remove the library the Web build and runs. 
Is anyone else having the same issue?  I was using the Expo v50 and just updated to the v51, but still have the issue.

![image](https://github.com/supabase/supabase/assets/107908549/bbe2614d-9705-42bf-bb00-5aedaebfa5b1)

**Comment by y7ya-com** [comment]
Did you find a solution for this? Running into the same problem.

**Comment by Lewitje** [comment]
Same issue here

**Comment by John7bigo** [comment]
@yayashn @Lewitje 

A quick workaround will be to disable the persistSession prop to false. And then you'll be able to launch the expo web app 

‐-------------------------------------------
import 'react-native-url-polyfill/auto'
import { createClient } from '@supabase/supabase-js'
import AsyncStorage from "@react-native-async-storage/async-storage";

const supabase = createClient("https://xyzcompany.supabase.co", "public-anon-key", {
  auth: {
    storage: AsyncStorage,
    autoRefreshToken: true,
    persistSession: false, // set to false here. 
    detectSessionInUrl: false,
  },
});

**Reply by oussamabenkortbi** [reply]
`persistSession: Platform.OS !== "web"`
solved the issue for me, but i want a better solution

**Reply by DaveyEke** [reply]
I did this persistSession: Platform.OS === "web" ? false : true,

**Comment by XFRNO** [comment]
EDIT: CHECK https://github.com/supabase/supabase-js/issues/870 FOR A BETTER SOLUTION

in app.json change the output value from static to single

```
"web":  {
	"output": "single"
  },
```

**Reply by naqeebali-shamsi** [reply]
This increases the bundle size though.

**Reply by matheusbaumgart** [reply]
Any other solution?

**Reply by kaminskypavel** [reply]
wouldn't guess in a million years! thank 🙏

**Reply by zhtgeneral** [reply]
God

**Reply by arthurvaverko** [reply]
how in the world is this related ???? can soemone explain to me ... i was frustrated fro 2 days until i hit this !

**Comment by champec** [comment]
anyone get a solution?

**Comment by partytoni** [comment]
Is there in supabase any mechanism that let us resume the session given the access-token of the previous session?

With this bug there is currently no way for the developers to make the web UX streamlined like on Android and iOS.

Please solve this issue!

**Comment by fab-uleuh** [comment]
https://github.com/supabase/supabase-js/issues/870

**Reply by partytoni** [reply]
For everyone interested, in this issue there's the solution. Thank you @fab-uleuh
